### PR TITLE
Added custom parameter for apply_service

### PR DESCRIPTION
**Pull request by jkroepke**
Hi,

I want use services like this:

```
apply Service "check_http" for (name => config in host.vars.http) {
      import "generic-service"
  display_name = "HTTP " + name + ""
  assign where "linux" in host.groups
  check_command = "check_http"
  vars += config
}
```

Currently, the line "vars += config" is impossible to set via puppet.

### CODE REVIEW

**Comment by lazyfrosch** [comment]
I like the idea, will try it out and merge.

Could you provide a simple spec test?

File: spec/defines/icinga2_object_apply_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      it { should_not contain_file(object_file).with_content(/^\s*check_command = "check_http"/) }
+
+    end
+    context "on #{name} with custom_prepend parameter in string format" do
+      let :facts do
+        facts
+      end
+      let :params do
+      {
+        custom_prepend: => 'vars += config',
```
Comment (lazyfrosch) [inline spec/defines/icinga2_object_apply_service_spec.rb]: Shouldn't commend without reading closely :tired_face:

`:name => 'value',`

**Comment by lazyfrosch** [comment]
You can run the tests locally, e.g.

``` bash
rake spec_prep
rspec spec/defines/icinga2_object_apply_service_spec.rb
# or
rake spec
```

**Comment by jkroepke** [comment]
Local test do not helped me. This is the result, if I run local tests:

679 examples, 149 failures, 11 pending

Anyway, I wrote working tests.

**Comment by lazyfrosch** [comment]
@jkroepke thanks for the awesome test! Only one remark, usually you never test for breakage, just for the sake of the validate_\* command to work :)

**Pull request merged**